File: Algorithms/Game Theory/Misère Nim.py
import os

#
# Complete the 'misereNim' function below.
#
# The function is expected to return a STRING.
# The function accepts INTEGER_ARRAY s as parameter.
#

def misereNim(s):
    # Write your code here
    nim_sum = 0
    for pile in s:
        nim_sum ^= pile

    if all(pile == 1 for pile in s):
        return "First" if len(s) % 2 == 0 else "Second"

    return "Second" if nim_sum == 0 else "First"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input().strip())

    for t_itr in range(t):
        n = int(input().strip())

        s = list(map(int, input().rstrip().split()))

        result = misereNim(s)

        fptr.write(result + '\n')

    fptr.close()

File: Algorithms/Game Theory/Misère Nim.php
<?php

/*
 * Complete the 'misereNim' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER_ARRAY s as parameter.
 */

function misereNim($s) {
    # Write your code here
    $nim_sum = 0;
    foreach($s as $pile){
        $nim_sum ^= $pile;
    }

    if (array_reduce($s, function($carry, $pile) { return $carry && $pile == 1; }, True))
        return (count($s) % 2 == 0) ? "First" : "Second";

    return ($nim_sum == 0) ? "Second" : "First";
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$t = intval(trim(fgets(STDIN)));

for ($t_itr = 0; $t_itr < $t; $t_itr++) {
    $n = intval(trim(fgets(STDIN)));

    $s_temp = rtrim(fgets(STDIN));

    $s = array_map('intval', preg_split('/ /', $s_temp, -1, PREG_SPLIT_NO_EMPTY));

    $result = misereNim($s);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
